jupyterlab_latex/__init__.py

```diff
@@ -159,39 +159,40 @@ def run_latex(self, command_sequence):
         command_sequence : list of tuples of strings
             This is a sequence of tuples of strings to be passed to
             `tornado.process.Subprocess`, which are to be run sequentially.
+            On Windows, `tornado.process.Subprocess` is unavailable, so
+            we use the synchronous `subprocess.run`.
 
         Returns
         -------
         string
             Response is either a success or an error string.
 
-        Raises
-        ------
-        tornado.process.CalledProcessError
-
         Notes
         -----
         - LaTeX processes only print to stdout, so errors are gathered from
           there.
 
         """
+        # Windows does not support async subprocesses, so
+        # use a synchronous system calls.
+        if sys.platform == 'win32':
+            run_command = run_command_sync
+        else:
+            run_command = run_command_async
+
         for cmd in command_sequence:
-            process = Subprocess(cmd, 
-                                 stdout=Subprocess.STREAM,
-                                 stderr=Subprocess.STREAM)
-            try:
-                yield process.wait_for_exit()
-            except CalledProcessError as err:
+            code, output = yield run_command(cmd)
+            if code != 0:
                 self.set_status(500)
                 self.log.error((f'LaTeX command `{" ".join(cmd)}` '
                                  'errored with code:\n ')
-                               + str(err.returncode))
-                out = yield process.stdout.read_until_close()
-                return out
+                               + str(code))
```

Comment:
why don't we just stick with the f-string pattern (so f'{code}' in this case rather than appending)